apps/cli/src/lib/git/diff.ts

```diff
@@ -67,3 +47,19 @@ export function getDiff(left: string, right: string | undefined): string {
     resource: 'getDiff',
   });
 }
+
+function hasDiff(opts: { args: string[]; resource: string }): boolean {
+  try {
+    runGitCommand({
+      args: ['diff', '--no-ext-diff', '--quiet', ...opts.args],
+      onError: 'throw',
+      resource: opts.resource,
+    });
+  } catch (e) {
+    if (e instanceof CommandFailedError && e.status === 1) {
+      return true;
+    }
```

Comment:
**question:**  Can you say more about this error catching? It seems wrong/counter-intuitive but I assume it's here for a reason. What is `exit 1` in this case?

**chore (non-blocking):** Can you add a comment here explaining what this is?

Reply:
The exit codes are documented in the man page: https://git-scm.com/docs/git-diff#Documentation/git-diff.txt---exit-code

Would you rather the exit code is returned from `runGitCommand` when the exit code is expected? I don't think there's much more comment that can be left here which doesn't just repeat the code.

Reply:
I think a comment is helpful:

- It's unclear to the average user that they'd have to find the specific `man` entry about `--exit-code`
- No one wants to dive around in `man` pages, this is more immediate & clear
- Exit codes feel arcane in general

```suggestion
    // When ran with `--exit-code`, if there are any differences, the `git` program exits with code `1`.
    // Otherwise, it exits with `0`.
    if (e instanceof CommandFailedError && e.status === 1) {
      return true;
    }
```